pos_lot_selection/static/src/js/popup/OrderWidget.js

```diff
@@ -0,0 +1,47 @@
+odoo.define('pos_lot_selection.CustomOrderWidget', function(require) {
+    'use strict';
+
+    const Registries = require('point_of_sale.Registries');
+    const OrderWidget = require('point_of_sale.OrderWidget');
+
+    
+    var CustomOrderWidget = (OrderWidget) => class extends OrderWidget{
+        async _editPackLotLines(event) {
+            const orderline = event.detail.orderline;
+            
+            this.env.session.lots = await this.get_lots(orderline.product, this.env.session);
+            return super._editPackLotLines(event);
+        }
+
+        async get_lots(product){
```

Comment:
I guess the best place where to define this function is in the POS DB, so that you could call it using `this.env.pos.db.get_lots` in both cases.